plugins/chat/assets/javascripts/discourse/services/chat-composer-warnings-tracker.js

```diff
@@ -63,61 +58,42 @@ export default class ChatComposerWarningsTracker extends Service {
       return;
     }
 
-    const mentions = this._extractMentions(currentMessage.message);
-    this.mentionsCount = mentions?.length;
-
-    if (this.mentionsCount > 0) {
-      this.tooManyMentions =
-        this.mentionsCount > this.siteSettings.max_mentions_per_chat_message;
+    currentMessage.parseMentions().then((mentions) => {
```

Comment:
After the last commits, this line is the only significant change to this function, we called `this._extractMentions` before, now we call `currentMessage.parseMentions()`.